### PR TITLE
fix: fix the CEL definitions to allow policies that use target selectors without explicit targetRefs

### LINKED ISSUE #3901: Creation of SecurityPolicy with `targetSelectors` fails
*Description*:

> Creation of SecurityPolicy with `targetSelectors` only fails with the `either targetRef or targetRefs must be used` message

*Repro steps*:
> Create SecurityPolicy below

```
apiVersion: gateway.envoyproxy.io/v1alpha1
kind: SecurityPolicy
metadata:
  labels:
    eg/namespace: reference-apps
  name: eg-fallback-oidc
  namespace: reference-apps
spec:
  oidc:
    clientID: xxxxxxxxxx
    clientSecret:
      name: oauth2-secret
    provider:
      issuer: >-
        https://login.microsoftonline.com/xxxxxxxxxx/v2.0
    scopes:
      - 'api://xxxxxxxxxx/oidc'
      - offline_access
  targetSelectors:
    - group: gateway.networking.k8s.io
      kind: HTTPRoute
      matchLabels:
        eg/namespace: reference-apps
```

*Environment*:

> Version v1.1.0-rc.1, installed with the helm chart

```
dependencies:
- name: gateway-helm
  version: v1.1.0-rc.1
  repository: oci://docker.io/envoyproxy
```

It must be coming from the [x-kubernetes-validation](https://github.com/envoyproxy/gateway/blob/main/charts/gateway-helm/crds/generated/gateway.envoyproxy.io_securitypolicies.yaml#L1131-L1133). 

These CRDS are in the `generated` directory and I'm failing to find their source to fix the bug

### PR DESCRIPTION
**Pull request by liorokman**
Fixes #3901

### CODE REVIEW

**Review by guydc — APPROVED** [review]
LGTM, Thanks!

File: api/v1alpha1/clienttrafficpolicy_types.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ type ClientTrafficPolicy struct {
 	Status gwapiv1a2.PolicyStatus `json:"status,omitempty"`
 }
 
-// +kubebuilder:validation:XValidation:rule="(has(self.targetRef) && !has(self.targetRefs)) || (!has(self.targetRef) && has(self.targetRefs))", message="either targetRef or targetRefs must be used"
+// +kubebuilder:validation:XValidation:rule="(has(self.targetRef) && !has(self.targetRefs)) || (!has(self.targetRef) && has(self.targetRefs)) || (has(self.targetSelectors) && self.targetSelectors.size() > 0) ", message="either targetRef or targetRefs must be used"
```
Comment (arkodg) [inline api/v1alpha1/clienttrafficpolicy_types.go:37]: does this mean a user can specify targetRefs and targetSelectors at the same time ?

**Review by Xunzhuo — APPROVED** [review]
Good catch

File: api/v1alpha1/clienttrafficpolicy_types.go (hunk 1 above)
Comment (liorokman) [reply api/v1alpha1/clienttrafficpolicy_types.go:37]: Yes.

**Comment by sadovnikov** [comment]
Thanks! 👍

**Pull request merged**